src/main/java/com/google/firebase/auth/UserRecord.java

```diff
@@ -569,7 +604,24 @@ UpdateRequest setValidSince(long epochSeconds) {
       }
 
       if (copy.containsKey("phoneNumber") && copy.get("phoneNumber") == null) {
-        copy.put("deleteProvider", ImmutableList.of("phone"));
+        Object deleteProvider = copy.get("deleteProvider");
+        if (deleteProvider != null) {
+          if (!(deleteProvider instanceof Iterable<?>)) {
+            throw new IllegalStateException("Contents of deleteProvider was not iterable");
+          }
+
+          // Due to java's type erasure, we can't fully check the type. :(
+          @SuppressWarnings("unchecked")
+          Iterable<String> deleteProviderIterable = (Iterable<String>)deleteProvider;
+
+          copy.put("deleteProvider", new ImmutableList.Builder<String>()
```

Comment:
Should we handle the double specification case?

```
user.updateRequest()
  .setPhoneNumber(null)
  .setProvidersToUnlink(ImmutableList.of("phone"))
```

I remember handling this case is other languages.

Reply:
Would it make sense to enforce this constraint in the setters? We'd have to do so in both `setPhoneNumber()` and `setProvidersToUnlink()`, but I think we can reduce the duplication via some sort of a helper function. That would make argument validation consistent with the pattern currently used in this class.

Reply:
Yeah, that's better; done. I've removed the logic here, i.e. if we somehow get into a bad state, we won't notice it. (But now we shouldn't get into that state.)

I didn't create a helper; there wasn't much duplication (though the exception itself is duplicated). If we set the value first and then checked, it would work out better... but that risks leaving the object in a bad state (eg if user catches and ignores the exception and then proceeds to use the object anyways.)